src/main/java/org/localhost/pizzeria/supplies/system/service/impl/SupplySystemCommandServiceImpl.java

```diff
@@ -1,25 +1,97 @@
     }
 
     @Override
     public void updateIngredientRestockStatus(long ingredientId) {
-        Ingredient ingredient = systemRepository.findById(ingredientId)
+        Ingredient ingredient = supplySystemRepository.findById(ingredientId)
                 .stream()
                 .findFirst()
                 .orElseThrow(
                         () -> new IngredientNotFoundException(IngredientExceptionsDetails.INGREDIENT_NOT_FOUND)
                 );
     }
+
+    @Override
+    @Transactional(rollbackFor = Exception.class)
+    public void decreaseIngredientStock(long ingredientId, int amount) {
+        Ingredient ingredient = getIngredient(ingredientId);
+        ingredient.setAmountInStock(ingredient.getAmountInStock() - amount);
+        supplySystemRepository.save(ingredient);
+    }
+
+    @Override
+    public void increaseIngredientStock(long ingredientId, int amount) {
+        Ingredient ingredient = getIngredient(ingredientId);
+        ingredient.setAmountInStock(ingredient.getAmountInStock() + amount);
+        supplySystemRepository.save(ingredient);
+    }
+
+    @Override
+    @Transactional(rollbackFor = Exception.class)
+    public IngredientDTO registerNewIngredient(NewIngredientDto ingredientData) {
+        if (Objects.isNull(ingredientData)) {
+            throw new IllegalArgumentException("Ingredient data is null");
+        }
+        Ingredient ingredient = Ingredient.builder()
+                .productName(ingredientData.getProductName())
+                .unitType(ingredientData.getUnitType())
+                .minimumRequiredAmount(ingredientData.getMinimumRequiredAmount())
+                .amountInStock(ingredientData.getAmountInStock())
+                .build();
+        Ingredient savedIngredient = supplySystemRepository.save(ingredient);
+
+        return IngredientDTO.fromIngredient(savedIngredient);
+    }
+
+    @Override
+    public IngredientDTO updateIngredientData(long ingredientId, EditIngredientDto ingredientData) {
+        if (Objects.isNull(ingredientData)) {
+            throw new IllegalArgumentException("Ingredient data is null");
+        }
+        Ingredient ingredient = getIngredient(ingredientId);
+        ingredient.setProductName(ingredientData.getProductName());
+        ingredient.setMinimumRequiredAmount(ingredientData.getMinimumRequiredAmount());
+
+        Ingredient updatedIngredient = supplySystemRepository.save(ingredient);
+        return IngredientDTO.fromIngredient(updatedIngredient);
+    }
+
+    @Override
+    public void removeIngredient(long ingredientId) {
+        if (ingredientId <= 0) {
+            throw new IllegalArgumentException("Ingredient id must be greater than 0");
+        }
+        Ingredient ingredient = getIngredient(ingredientId);
+        supplySystemRepository.delete(ingredient);
+    }
+
+    @Override
+    public List<Ingredient> submitOrderToStockQueue(List<Long> ingredientIds) {
```

Comment:
jezeli nie masz na razie implementacji dla danego kodu to albo go nie implementuj albo rzucaj wyjatek